src/PIL/GifImagePlugin.py

```diff
@@ -457,7 +457,9 @@ def _save_all(im, fp, filename):
 
 
 def _save(im, fp, filename, save_all=False):
-    im.encoderinfo.update(im.info)
+    for k, v in im.info.items():
+        if k not in im.encoderinfo:
+            im.encoderinfo[k] = v
```

Comment:
One line instead of two:
```python
im.encoderinfo.setdefault(k, v)
```

Reply:
Okay, thanks, I've updated the commit.

Reply:
I've been writing python for 15 years and I've never seen that.